gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -325,12 +325,14 @@ public Spec getSpecWrapper(URI uri) {
     return responseMap;
   }
 
-  private boolean isCompileSuccessful(Map<String, AddSpecResponse> responseMap) {
+  public static boolean isCompileSuccessful(Map<String, AddSpecResponse> responseMap) {
     AddSpecResponse<String> addSpecResponse = responseMap.getOrDefault(ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(""));
 
-    return addSpecResponse != null
-        && addSpecResponse.getValue() != null
-        && !addSpecResponse.getValue().contains("ConfigException");
+    return isCompileSuccessful(addSpecResponse.getValue());
+  }
+
+  public static boolean isCompileSuccessful(String dag) {
```

Comment:
Can we move this method into BaseFlowCompiler class and expose the method via compiler interface? It seems like a property of the compiler. Definitely, not that of FlowCatalog.

Reply:
Yes, we can move the method into gobblin-service module, but gobblin-runtime (FlowCatalog) cannot have dependency on gobblin-service. That means listeners have to return a map in AddSpecResponse if they want to pass multiple information ( a) isCompileSuccessful b) compiledDag) to the caller.
Let me know if you want me implement it that way.

Reply:
Understood. One option would be to relocate FlowCatalog inside gobblin-service module. Not sure, if it will require a significant refactoring effort, in which case, it is ok to leave it as-is and we can do the refactoring in a separate PR.